siibra/Region.m

```diff
@@ -0,0 +1,71 @@
+classdef Region
+    properties
+        name
+        id
+        parcellation
+        space_url
+    end
+    methods
+        function region = Region(name, id, parcellation, dataset_specs)
+            region.name = name;
+            region.id = id;
+            region.parcellation = parcellation;
+
+            % parse dataset_specs for this region
+            if ~isempty(dataset_specs)
+                for i = 1:numel(dataset_specs)
+                    if iscell(dataset_specs)
+                        space_id = dataset_specs{i, 1}.space_id;
+                        url = dataset_specs{i, 1}.url;
+                    else
+                        space_id = dataset_specs(i).space_id;
+                        url = dataset_specs(i).url;
+                    end
+                    % currently "MNI152 2009c nonl asym" is supported only 
+                    if space_id == "minds/core/referencespace/v1.0.0/dafcffc5-4826-4bf1-8ff6-46b8a31ff8e2"
+                        region.space_url = url;
+                        break;
+                    end
+                end
+            else
+                % if no region map available
+                region.space_url = "";
+            end
+        end
+        function children = getChildrenNames(obj)
+            children = obj.parcellation.getChildrenNames(obj.name);
+        end
+        function pmap = probabilityMap(obj)
+            if obj.space_url == ""
+                error("This region has no region map!");
+            end
+            nifti_data = api_call(obj.space_url);
+            file_handle = fopen("tmp_nifti.nii.gz", "w");
```

Comment:
Suggest to use `niftiwrite` if possible. Is there a blocker you encountered?

Reply:
The api returns compressed nifti. As far as I understand, nifitwrite does not support writing already compressed data. So calling `niftiwrite` and then `niftiread` did'nt work out. This is why I resorted to standard fopen, fwrite, ...